src/java/org/apache/cassandra/cql3/transactions/ConditionStatement.java

```diff
@@ -100,12 +100,18 @@ public ConditionStatement prepare(String keyspace, VariableSpecifications bindVa
             
             if (lhs instanceof RowDataReference.Raw)
             {
+                if (((RowDataReference.Raw) lhs).column() == null) {
+                    throw new IllegalStateException(String.format("Row reference (%s) can only be used with IS NULL/IS NOT NULL conditions", lhs.getText()));
```

Comment:
this check isn't correct

`row IS NULL` and `row IS NOT NULL` both enter this block and this check will now reject that.  We need to support checking row nullability.

Reply:
@dcapwell  Wouldn't IS NULL/ IS NOT NULL gets returned earlier itself? 
In [this](https://github.com/apache/cassandra/blob/52e982389c9a2bc1c80f4473480a072bad6e78c4/src/java/org/apache/cassandra/cql3/transactions/ConditionStatement.java#L90C14-L91C1) snippet, we check if rhs is NULL. 

Wouldn't IS NULL/ IS NOT NULL  go into this IF block?